river/anomaly/gaussian.py

```diff
@@ -64,10 +71,24 @@ def _dist(self):
         )
 
     def learn_one(self, _, y):
-        self.gaussian.update(y)
+        if not self.grace_period_passed and self.grace_period:
+            self.grace_period_counter += 1
+            if self.grace_period_counter > self.grace_period: 
+                self.grace_period_passed = True
+        if isinstance(self.gaussian, proba.TimeRolling):
+            t = _[self.timestamp_label] if isinstance(_, dict) else _
+            parsed_t = (
+                t if isinstance(t,dt.datetime)
+                else dt.datetime.utcfromtimestamp(t)  if isinstance(t,int)
+                else dt.datetime.strptime(t,self.timestamp_format)
+            )
+            parsed_t =  parsed_t if parsed_t.tzinfo else pytz.utc.localize(parsed_t)
+            self.gaussian.update(y, parsed_t)
+        else:
+            self.gaussian.update(y)
         return self
 
     def score_one(self, _, y):
-        if self._dist.n_samples < self.grace_period:
+        if self._dist.n_samples < self.grace_period and not self.grace_period_passed:
```

Comment:
Sorry but I don't understand. In what cases is `and not self.grace_period_passed` bringing something to the table? Isn't checking for `self._dist.n_samples < self.grace_period` sufficient?

Reply:
well if you have a grace period of 200 and a time window of 30, then the condition `self._dist.n_samples < self.grace_perion` will never get true, and thus the model will never update